File: backend/views/folder.py
from utils.folder import add_folder_util, get_folders_util, delete_folder

def add_folder(request):
    add_folder_resp = add_folder_util(request)

    if "error" in add_folder_resp:
        add_folder_resp['success'] = False
    else:
        add_folder_resp['success'] = True
        add_folder_resp['status_code'] = 200

    return add_folder_resp

def get_folders_view(request):
    folder_data = get_folders_util(request)
    if "error" in folder_data:
        folder_data['success'] = False
    else:
        folder_data['success'] = True
        folder_data['status_code'] = 200
    return folder_data

def delete_folder_view(request):
    result = delete_folder(request)
    if "error" in result:
        result['success'] = False
    else:
        result['success'] = True
        result['status_code'] = 200
    return result

File: frontend/src/Components/Folder.js
import React from 'react'
import '../Stylesheets/Folder.css'
import { useStateValue } from '../Data/StateProvider';
import DeleteOutlineIcon from '@material-ui/icons/DeleteOutline';
import $ from 'jquery';

import client from '../services/Client'

function Folder({a_folder}) {

    const [{active_user, active_class, active_nav, refresh}, dispatch] = useStateValue();


    const refresher = () => {
        dispatch({
            type: 'SET_REFRESH',
            item: !(refresh),
        })
    }
    const setFolder =(folder_dict) => {
        console.log(folder_dict);
        dispatch({
            type: 'ADD_ACTIVE_FOLDER',
            item: folder_dict
        })
    }

    const deleteFolder = (id) => {
        $.ajax({
            url: `/folders?id=${id}&class=${active_class}&category=${active_nav}&shared_to=${active_user}&user=${client.getUserData()}`, //TODO: update request URL
            type: "DELETE",
            dataType: 'json',
            contentType: 'application/json',
            xhrFields: {
              withCredentials: true
            },
            crossDomain: true,
            success: (result) => {
              console.log(result)
              refresher();
              return;
            },
            error: (error) => {
              // console.log(error)
              // alert(error.responseJSON.message)
              refresher();
              return;
            }
        })
    }

    return (
        <div className="folder">
            <div onClick={()=>{deleteFolder(a_folder.id)}} className="folder__delete-buttone">
                <DeleteOutlineIcon />
            </div>
            <div onClick={()=>{setFolder(a_folder)}} className="folder__content">
                <img src="../images/folder.png" alt="folder" />
            </div>
            <div className="folder__title">
            {a_folder.folder}
            </div>
        </div>
    )
}

export default Folder
